Realm/Realm.Unity/Editor/UnityWeaver.cs

```diff
@@ -103,35 +105,64 @@ private static async Task WeaveAssembly(string assemblyPath)
                 var timer = new Stopwatch();
                 timer.Start();
 
-                var (moduleDefinition, fileStream) = WeaverAssemblyResolver.Resolve(assembly);
-                if (moduleDefinition == null)
+                var resolutionResult = WeaverAssemblyResolver.Resolve(assemblyPath, references);
+                if (resolutionResult == null)
                 {
                     return;
                 }
-                using (fileStream)
-                using (moduleDefinition)
+
+                using (resolutionResult)
                 {
                     // Unity doesn't add the [TargetFramework] attribute when compiling the assembly. However, it's
                     // using NETStandard2, so we just hardcode this.
-                    var weaver = new Weaver(moduleDefinition, logger, new FrameworkName(".NETStandard,Version=v2.0"));
+                    var weaver = new Weaver(resolutionResult.Module, logger, new FrameworkName(".NETStandard,Version=v2.0"));
                     var results = weaver.Execute();
 
-                    moduleDefinition.Write(WriterParameters);
-
                     // Unity creates an entry in the build console for each item, so let's not pollute it.
                     if (results.SkipReason == null)
                     {
+                        resolutionResult.SaveModuleUpdates();
                         logger.Info($"[{name}] Weaving completed in {timer.ElapsedMilliseconds} ms.{Environment.NewLine}{results}");
                     }
                 }
-
-                // save any changes to our weavedAssembly objects
-                // TODO: verify we need to do this
-                AssetDatabase.SaveAssets();
             }
             catch (Exception ex)
             {
-                logger.Warning($"[{name}] Weaving failed: {ex.Message}");
+                logger.Warning($"[{name}] Weaving failed: {ex.StackTrace}");
+            }
+        }
+
+        public void OnPostBuildPlayerScriptDLLs(BuildReport report)
+        {
+            // This is a bit hacky - we need actual references, not directories, containing references, so we pass folder/dummy.dll
+            // knowing that dummy.dll will be stripped.
+            var systemAssemblies = CompilationPipeline.GetSystemAssemblyDirectories(ApiCompatibilityLevel.NET_Standard_2_0).Select(d => Path.Combine(d, "dummy.dll"));
+            var referencePaths = systemAssemblies
+                .Concat(report.files.Select(f => f.path))
```

Comment:
Yeah - that was a bit tricky to express in a readable way. Essentially, what happens is that we have two collections of paths:
1. System assembly directories - it contains all the **folders** where we have system assemblies
2. report.files.path - it contains the paths of all **files** that were compiled

These two are not at the same level - one is a folder, containing files, the other is a file. There were two approaches I could take to bring them to represent the same thing:
1. Get the directory containing the files in report.files.path - essentially doing something like `report.files.Select(f => Path.GetDirectoryName(f.path))`. This would give us the parent folder for each file (duplicates are fine).
2. Add a dummy file to the system directories, to make them look like filepaths.

I chose option 2 because it worked better with the weaver invocation by the CompilationPipeline flow and reduced the amount of code needed.